### PR TITLE
Speed up time to retrieve reads from BAM files

### PR DESCRIPTION
**Pull request by robertaboukhalil**
This PR optimizes fetching reads from BAM files and should help prevent the app from crashing when trying to load in too much data:

1. Instead of using `samtools view`, which appends each line to Aioli's `STDOUT` variable, we now use `samtools view -o /tmp/reads.sam`, which sends the output to a temporary file on the virtual file system, and we `cat` the file back at the end. This introduces ~2-3X speedup simply because we do the bytes --> string conversions only once with a large payload instead of doing it after every single break line.
2. Run `samtools coverage` before trying to fetch the data. If the user is trying to fetch too much data, warn them and suggest how much downsampling samtools should do to keep the app responsive.

### CODE REVIEW

File: js/vis.js [hunk 1]
```diff
@@ -4431,40 +4431,18 @@ function open_bam_file(event) {
 document.getElementById('bam_file').addEventListener('change',open_bam_file,false);
 
 function create_bam(files) {
```
Comment (robertaboukhalil) [inline js/vis.js:4433]: Simplify the logic for validating that the user specified a .bam and a .bai/.csi

File: js/vis.js [hunk 2]
```diff
@@ -4714,34 +4692,6 @@ function parse_bam_record(record) {
 
 }
 
-function parse_bam_text(bam_text) {
```
Comment (robertaboukhalil) [inline js/vis.js:4717]: This function isn't used

File: js/vis.js [hunk 3]
```diff
@@ -5161,8 +5111,14 @@ class Bam
 	{
 		// Get reads from file in the browser
 		if(this.source == "file") {
```
Comment (robertaboukhalil) [inline js/vis.js:5113]: This is the crux of the PR: ask samtools to output the SAM file to a file, then use Aioli's `cat` feature to retrieve the file contents

File: js/vis.js [hunk 4]
```diff
@@ -5235,8 +5191,8 @@ class Bam
 					cigar: readInfo[5]
 				};
```
Comment (robertaboukhalil) [inline js/vis.js:5193]: Minor fix: SA tags start at the 12th column, so we can start looking there

File: package.json [hunk 5]
```diff
@@ -7,12 +7,12 @@
     "url": "git+https://github.com/MariaNattestad/Ribbon.git"
   },
   "scripts": {
```
Comment (robertaboukhalil) [inline package.json:9]: Update Aioli to use the new @biowasm org

File: js/vis.js [hunk 6]
```diff
@@ -5235,8 +5215,8 @@ class Bam
...
 
-				// Parse SA tag
-				for (var i = 0; i < readInfo.length; i++) {
+				// Parse SA tag: In the SAM format, column 11 = readInfo[10] = qual, column 12 = readInfo[11] = start of tags
```
Comment (MariaNattestad) [inline js/vis.js:5218]: This was a little hard to read, but I think you mean the tags start at index 11, right?

**Comment by robertaboukhalil** [comment]
Added a fix so that if users are running Ribbon in automation mode, there won't be an alert box asking about subsampling. Also added an option in the advanced settings so users can turn off subsampling during automation entirely.

**Review by MariaNattestad — APPROVED** [review]

**Pull request merged**